### PR TITLE
Add CELU activation to pytorch

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Also fuse input scale multiplication into ELU

Paper:
https://arxiv.org/pdf/1704.07483.pdf

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
Can any comment on the error for the gcc7.2 build?
```
error: '<anonymous>.at::Scalar::v' may be used uninitialized in this function [-Werror=maybe-uninitialized]
```
I don't have a machine with gcc-7 for now, so it's hard for me to look into the error.

**Comment by zasdfgbnm** [comment]
The test for
```
continuous-integration/travis-ci/pr
pr/pytorch-linux-trusty-py3.5
pr/caffe2-py2-gcc5-ubuntu16.04-test
```
seems to be false positive

**Comment by zasdfgbnm** [comment]
Just one unrelated fail, should be working now.

**Comment by zasdfgbnm** [comment]
Just some information: Changes in this PR is very similar to https://github.com/pytorch/pytorch/pull/4269 and https://github.com/pytorch/pytorch/commit/b6a30f7edefe86faff570680fd2a668315c1da18 by @colesbury

**Comment by SsnL** [comment]
Thanks for your contribution! Were you able to compile locally? It seems to fail to compile on our CI machines.

File: aten/src/ATen/native/Activation.cpp [hunk 1]
```diff
@@ -25,6 +25,16 @@ Tensor & selu_(Tensor & self) {
   return at::elu_(self, SELU_ALPHA, SELU_SCALE);
 }
 
+Tensor celu(const Tensor & self, Scalar alpha) {
+  Tensor inv_alpha = alpha.toTensor().reciprocal();
```
Comment (SsnL) [inline aten/src/ATen/native/Activation.cpp]: Can you instead do `double inv_alpha = 1. / alpha.to<double>()`? This avoids launching functions optimized for large tensors.

**Review by houseroad — COMMENTED** [review]
ONNX part LG to me.

Nit: add test case in https://github.com/pytorch/pytorch/blob/master/test/onnx/test_operators.py to cover Elu

File: torch/nn/modules/activation.py [hunk 2]
```diff
@@ -299,6 +300,43 @@ def extra_repr(self):
         return 'alpha={}{}'.format(self.alpha, inplace_str)
 
 
+class CELU(Module):
+    r"""Applies element-wise,
+    :math:`\text{CELU}(x) = \max(0,x) + \min(0, \alpha * (\exp(x/\alpha) - 1))`
+
+    More details can be found in the paper `Continuously Differentiable Exponential Linear Units`_ .
```
Comment (SsnL) [inline torch/nn/modules/activation.py]: This paper is missing a link. Here is an example on how to add it: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L151-L152

**Comment by zasdfgbnm** [comment]
@houseroad @SsnL I've made those improvements.

File: torch/nn/modules/activation.py (hunk 2 above)
Comment (zasdfgbnm) [reply torch/nn/modules/activation.py]: fixed

**Comment by SsnL** [comment]
Thanks! @zasdfgbnm
